internal/api/shared/rbac/cron.go

```diff
@@ -15,21 +15,53 @@ package rbac
 
 import (
 	"context"
+	"time"
 
 	"github.com/perses/common/async"
+	"github.com/perses/perses/internal/api/config"
+	"github.com/perses/perses/internal/api/shared/database/model"
+	"github.com/sirupsen/logrus"
 )
 
-func NewCronTask(rbacService RBAC) async.SimpleTask {
-	return &rbacTask{svc: rbacService}
+func NewCronTask(rbacService RBAC, dbConf config.Database, persesDAO model.DAO) async.SimpleTask {
+	return &rbacTask{svc: rbacService, dbConf: dbConf, persesDAO: persesDAO}
 }
 
 type rbacTask struct {
 	async.SimpleTask
-	svc RBAC
+	svc       RBAC
+	dbConf    config.Database
+	persesDAO model.DAO
 }
 
+// Execute checks every X seconds: if role, globalrole, rolebinding or globalrolebinding tables have been updated
+// If yes, refresh rbac cache
+// TODO: support diff database timezone
 func (r *rbacTask) Execute(_ context.Context, _ context.CancelFunc) error {
-	return r.svc.Refresh()
+	if r.dbConf.SQL != nil {
+		lastUpdateTime, err := r.persesDAO.CheckTablesLatestUpdateTime([]string{"role", "rolebinding", "globalrole", "globalrolebinding"})
+		if err != nil {
+			return err
+		}
+
+		// TODO: check
+		if lastUpdateTime == nil {
+			return nil
+		}
+
+		lastRefreshTime, err := time.Parse("2006-01-02 15:04:05", *lastUpdateTime)
+		if err != nil {
+			return err
+		}
+
+		if r.svc.LastRefreshTime().Before(lastRefreshTime) {
```

Comment:
I would keep the time `lastRefreshTime` in the cron task as it is related to the goal of the cron task. Notion about lastRefreshTime doesn't change anything regarding how the rbac service is supposed to work